apps/google-analytics-4/frontend/src/hooks/useSidebarSlug/useSidebarSlug.tsx

```diff
@@ -5,22 +5,35 @@ import { ContentEntitySys, SidebarExtensionSDK } from '@contentful/app-sdk';
 import { pathJoin } from 'utils/pathJoin';
 import useGetFieldValue from '../useGetFieldValue';
 
+const SLUG_FIELD_INPUT_DELAY = 500;
+
 export const useSidebarSlug = (slugFieldInfo: ContentTypeValue) => {
-  const [isPublished, setIsPublished] = useState(false);
   const sdk = useSDK<SidebarExtensionSDK>();
 
   const { slugField, urlPrefix } = slugFieldInfo;
   const slugFieldValue = useGetFieldValue(slugField);
 
+  const [isPublished, setIsPublished] = useState(false);
+  const [debouncedSlugFieldValue, setDebouncedSlugFieldValue] = useState(slugFieldValue);
```

Comment:
just to confirm my assumption/understanding here –  we need a separate state vs just doing setSlugFieldValue in a debounced fashion because we need a real time state to populate the react component's field value, but a debounced one that is tied to a useEffect elsewhere that updates the chart?

if i'm on the right track here, I actually wonder why we're so eagerly re-rendering the chart in the first place? it could definitely just render with whatever the slug field has when it initializes and then have a refresh button to manually re-render it that's conditionally shown based on a diff state being detected. this could reduce the "magic" here quite a bit i suspect?

Reply:
I think it's a fair point btw, but for now I'd lean toward shipping this change and seeing if it doesn't make sense later to remove the "reactive" behavior of the chart altogether (or add a button to trigger a refresh etc)

Reply:
Yeah you got it, so we are essentially having to "debounce" the state updates caused by the `getFieldValue` hook changing with every key stroke. And you definitely bring up a good point, the app has been consistently updating with any update to the slug field, and we originally thought "oh thats neat" but it certainly caused these unnecessary re-renders. I think this fix might be worth adding for now, as its simple enough imo, but perhaps along the lines of refresh button, we only update the state when the user has 'published' the data, so published data is the single source of truth? If we feel something along those lines is necessary now I am fine implementing that as well 👍